third_party/gtest/CMakeLists.txt

```diff
@@ -1,21 +1,21 @@
-message(STATUS "Configuring and building gtest")
-if(MSVC)
-    # Force using shared runtime library when MSVC builds shared libraries
-    ginkgo_load_git_package(gtest_external
-        "https://github.com/google/googletest.git"
-        "6a7ed316a5cdc07b6d26362c90770787513822d4"
-        # Work around the linking errors when compiling gtest with CUDA
-        "-Dgtest_disable_pthreads=ON" "-Dgtest_force_shared_crt=ON")
-else()
-    ginkgo_load_git_package(gtest_external
-        "https://github.com/google/googletest.git"
-        "6a7ed316a5cdc07b6d26362c90770787513822d4"
-        # Work around the linking errors when compiling gtest with CUDA
-        "-Dgtest_disable_pthreads=ON"
-        "-DCMAKE_CXX_FLAGS=-fPIC")
+if(NOT GTest_FOUND)
+    message(STATUS "Fetching external GTest")
+    include(FetchContent)
+    FetchContent_Declare(
+        googletest
+        GIT_REPOSITORY https://github.com/google/googletest.git
+        GIT_TAG        6a7ed316a5cdc07b6d26362c90770787513822d4
+    )
+    set(gtest_disable_pthreads ON)
+    if(MSVC)
+        set(gtest_force_shared_crt ON)
+    endif()
```

Comment:
should it be `googletest_gtest_disable_pthreads` and `googletest_force_shared_crt`?

Reply:
Where does that prefix come from?

Reply:
googletest  is from the name declaration of `FetchContent_Declare`
I found it from the example, so I guess they follow this way to distinguish the variable from different project? 
![image](https://user-images.githubusercontent.com/19565938/121016495-7b285b00-c79c-11eb-8444-3ddbedafecae.png)

Reply:
that is the actual name of the variable in protobuf: https://github.com/protocolbuffers/protobuf/blob/a1d8f8f96900468f09ced2ce9d23dea5a1bc070c/cmake/CMakeLists.txt#L273

Reply:
Ah, I see. Sorry for wrong statement.